src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -596,6 +597,9 @@ Pipe ReadFromMergeTree::readInOrder(
             actions_settings, block_size, reader_settings);
 
         processor->addPartLevelToChunk(isQueryWithFinal());
+        processor->addVirtualRowToChunk(part_with_ranges.data_part->getIndex(), part_with_ranges.ranges.front().begin);
```

Comment:
I would prefer to add a new processor on top, rather than modifying the reading processor code. It would be less intrusive. This processor initially would return the chunk with virtual row, and then just pass all the read blocks.

Probably it would require creating a new processor. I suppose we can inherit from IInflatingTransform but with enabled `can_generate` flag from the beginning.